File: GUI/FMLOGIN.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace GUI
{
    public partial class FMLOGIN : Form
    {
        public FMLOGIN()
        {
            InitializeComponent();
        }

        //Libreria del juego
        BLL.GENERALA _generalaBLL = new BLL.GENERALA();

        private void BCancelar_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void FMLOGIN_Load(object sender, EventArgs e)
        {
            //cargo una lista de usuarios registrados.
            CBUsr.DataSource = null;
            CBUsr.DataSource = _generalaBLL.ListarUSRregistrados();
            CBUsr.DisplayMember = "usuario";
        }

        private void BAceptar_Click(object sender, EventArgs e)
        {
            BE.USUARIO unUsuario = null;
            //chequeo de que tengo una seleccion
            if (CBUsr.SelectedIndex > -1)
            {
                //guardo el usuario seleccionado
                unUsuario = (BE.USUARIO)CBUsr.SelectedItem;
            }
            //verifico que el usuario esta seleccionado y la passwd esta escrita
            if (unUsuario != null && !string.IsNullOrWhiteSpace(TBPasswd.Text))
            {
                TBPasswd.BackColor = Color.White;
                //le agreo la passwd
                unUsuario.passwd = TBPasswd.Text;
                //compruebo el usuario y passwd en la DB
                unUsuario = _generalaBLL.logInUsuario(unUsuario);
                //si me devuelve al usuario procedo
                if (unUsuario != null)
                {
                    //agrego al usuario logueado al formulario base
                    FMCONTENEDOR.liUSRLogin.Add(unUsuario);
                    MessageBox.Show("Bienvenido {0}", unUsuario.usuario);
                    this.Close();
                }
                else
                {
                    MessageBox.Show("Usuario y contraseña incorrecta, por favor intentelo de nuevo");
                    TBPasswd.BackColor = Color.Beige;
                }
            }
            else
            {
                TBPasswd.BackColor = Color.Beige;
            }

        }

    }
}

File: Generala-MF/FMLOGIN.vb
Public Class FMLOGIN
    Dim _generalaBLL As New BLL.GENERALA


    Private Sub BCancelar_Click(sender As Object, e As EventArgs) Handles BCancelar.Click
        Me.Close()
    End Sub


    Private Sub FMLOGIN_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        'cargo una lista de usuarios registrados.
        CBUsr.DataSource = Nothing
        CBUsr.DataSource = _generalaBLL.ListarUSRregistrados
        CBUsr.DisplayMember = "usuario"


    End Sub

    Private Sub BAceptar_Click(sender As Object, e As EventArgs) Handles BAceptar.Click
        Dim unUsuario As BE.USUARIO = Nothing
        If CBUsr.SelectedIndex > -1 Then
            unUsuario = DirectCast(CBUsr.SelectedItem, BE.USUARIO)
        End If
        If unUsuario IsNot Nothing And Not String.IsNullOrWhiteSpace(TBPasswd.Text) Then
            TBPasswd.BackColor = Color.White
            unUsuario.passwd = TBPasswd.Text
            unUsuario = _generalaBLL.logInUsuario(unUsuario)
            If unUsuario IsNot Nothing Then
                FMCONTENEDOR.liUSRLogin.Add(unUsuario)
                MsgBox("Bienvenido " & unUsuario.usuario)
                Me.Close()
            Else
                MsgBox("Usuario y contraseña incorrecta, por favor intentelo de nuevo")
                TBPasswd.BackColor = Color.Beige
            End If
        Else
            TBPasswd.BackColor = Color.Beige
        End If
        

    End Sub
End Class